annotation-processor/src/main/java/org/hibernate/validator/ap/checks/ConstraintCheckFactory.java

```diff
@@ -55,7 +60,15 @@ public ConstraintCheckFactory(Types typeUtils, ConstraintHelper constraintHelper
 		fieldChecks = CollectionHelper.newHashMap();
 		fieldChecks.put(
 				AnnotationType.CONSTRAINT_ANNOTATION,
-				new SingleValuedChecks( new StaticCheck(), new TypeCheck( constraintHelper ) )
+				new SingleValuedChecks(
+						new StaticCheck(),
+						new TypeCheck( constraintHelper ),
+						new AnnotationParametersSizeLengthCheck( annotationApiHelper ),
+						new AnnotationParametersPatternCheck( annotationApiHelper ),
+						new AnnotationParametersScriptAssertCheck( annotationApiHelper ),
+						new AnnotationParametersDigitsCheck( annotationApiHelper ),
+						new AnnotationParametersDecimalMinMaxCheck( annotationApiHelper )
+				)
 		);
 		fieldChecks.put(
 				AnnotationType.MULTI_VALUED_CONSTRAINT_ANNOTATION,
```

Comment:
I don't quite remember, do we need to add the checks here, too? I.e. would they be invoked for the individual elements when using e.g. `@List.Size({...}`? It may work already, but probably a good idea to have a test for it.